### PR TITLE
make environment generic

### PR DESCRIPTION
**Pull request by fabrizio-m**

### CODE REVIEW

**Comment by fabrizio-m** [comment]
!ci-build-me

**Review by querolita — APPROVED** [review]
Very nice! I just had two doubts, which I am sure wouldn't work. But just asking for curiosity.

File: kimchi/src/circuits/expr.rs [hunk 1]
```diff
@@ -112,17 +112,39 @@ pub struct Environment<'a, F: FftField> {
     /// Lookup specific polynomials
     pub lookup: Option<LookupEnvironment<'a, F>>,
 }
-pub trait GetCol<'a, F: FftField, C: ColTrait> {
-    fn get_col(&self, col: &C) -> Option<&'a Evaluations<F, D<F>>>;
+pub trait EnvTrait<'a, F: FftField> {
+    type Col: ColTrait;
+    fn domain(&self) -> EvaluationDomains<F>;
+    fn constants(&self) -> &Constants<F>;
+    fn vanishes_on_last_4_rows(&self) -> &'a Evaluations<F, D<F>>;
```
Comment (querolita) [inline kimchi/src/circuits/expr.rs:119]: I believe this method should be `vanishing_on_last_n_rows`, which recently changed in develop -> https://github.com/o1-labs/proof-systems/blob/develop/kimchi/src/circuits/expr.rs

Perhaps you have to merge develop onto these branches to get the most recent version.

**Review by querolita — CHANGES_REQUESTED** [review]

**Comment by volhovm** [comment]
Closing, as per @fabrizio-m most of what is useful was already integrated into folding.

**Pull request closed without merging**